http/src/v0/refs.rs

```diff
@@ -370,11 +400,12 @@ fn handle_dagpb_not_found(
     at: Cid,
     data: &[u8],
     needle: String,
-    path: &IpfsPath,
+    last_segment: bool,
+    opts: &WalkOptions,
 ) -> Result<(Cid, Loaded, Vec<String>), WalkError> {
     use ipfs::unixfs::ll::dagpb::node_data;
 
-    if needle == "Data" && path.len() == 0 && path.follow_dagpb_data() {
+    if needle == "Data" && last_segment && opts.follow_dagpb_data {
```

Comment:
a nit that I seem to have missed the previous time: I'd put that `last_segment` check as the first one to have that ｍａｘｉｍｕｍ　ｓｐｅｅｄ when short-circuiting does its thing :smiley:

Reply:
```suggestion
    if opts.follow_dagpb_data && last_segment && needle == "Data" {
```

warp speed! though I'd expect rustc or llvm handle this, but perhaps not the compiler cannot make such decisions? it should see that these are `&self` and disjoint but it might not want to check that none of them have interior mutability...?

Reply:
> the compiler cannot make such decisions

I sure hope it doesn't, as changing the order could alter the logic (if e.g. one of the checks also modified its argument)

I mostly don't think it makes an actual issue in performance, but hey, it won't hurt